winbuild/build_prepare.py

```diff
@@ -421,25 +430,41 @@ def find_msvs():
     }
 
 
-def extract_dep(url, filename):
-    import tarfile
+def download_dep(url: str, file: str) -> None:
     import urllib.request
+
+    ex = None
+    for i in range(3):
+        try:
+            print(f"Fetching {url} (attempt {i + 1})...")
+            content = urllib.request.urlopen(url).read()
+            with open(file, "wb") as f:
+                f.write(content)
+            break
```

Comment:
```suggestion
            ex = None
            break
```
You've changed the `for`/`else` to "If `ex` is truthy, raise".

There's actually a difference here, that I presume was unintended. The `for`/`else` option only raises the error if it doesn't break after success. This version might fail on the first loop, then succeed on the second loop, but because `ex` is still populated, it will still raise afterwards.

If you prefer this style to `for`/`else`, I suggest inserting a `ex = None` before `break`.

Reply:
Yep, it was unintended. I originally refactored it to return instead of raising, but then put the `raise` back. So I've put the `for`/`else` back too. Thanks!